### ISSUE TITLE
RxJS 6.2.x trustedSubscriber._addParentTeardownLogic is not a function error

### ISSUE THREAD

**Issue by nabzzz** [ReactiveX/rxjs#4090]
We are developing Cordova Application using Bootstrap 4 and Angular 6.  We upgraded RxJS version to 6.3.1. When we launch the Application in Samsung Mobile Device, its stuck in Splash screen and we are not getting any error message in the browser console.

So when we reverted both rxjs and rxjs-compat to V6.0.0, 6.1.0, 6.2.x. We are getting **trustedSubscriber._addParentTeardownLogic is not a function.** Error.

**Ng Version:**

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.6.8
@angular-devkit/build-angular     0.6.8
@angular-devkit/build-optimizer   0.6.8
@angular-devkit/core              0.6.8
@angular-devkit/schematics        0.6.8
@angular/cli                      6.0.8
@angular/compiler-cli             6.0.7
@ngtools/webpack                  6.0.8
@schematics/angular               0.6.8
@schematics/update                0.6.8
rxjs                              6.2.2
typescript                        2.7.2
webpack                           4.8.3

**Error Log:**

**Reproduction**
We cannot share StackBlitz because our application is cordova based mobile app

**Environment**
Andriod, IOS
- RxJS version:
- 6.1.x, 6.2.x
<img width="762" alt="rxjs screenshot" src="https://user-images.githubusercontent.com/5388648/45034778-04535600-b076-11e8-84e9-f9d178467624.PNG">

**Comment by cartant** [comment]
See #4077.

**Comment by nabzzz** [comment]
Not seeing any solution for above problem in #4077.

**Comment by pertrai1** [comment]
@nabzzz right now it is difficult to reproduce using a simple boilerplate app. Working on reproducing on a large scale app today to find out what is happening - if I can.

**Comment by sureshdotariya** [comment]
Our application is a huge one around 45,000-50,000 lines of code. We started developing this application in Angular 2, then we upgraded to 4, 5 and 6.  We are facing this issue after Rxjs v6.3.1 got released. Previously it's working fine. We tried to downgrade Angular CLI to 6.0.7, 6.0.8 . Its n't fixing the problem.

**Comment by pertrai1** [comment]
@sureshdotariya what about when you go back a version in RxJS?

**Comment by sureshdotariya** [comment]
We downgraded to RxJS V6.0.0, 6.1.x, 6.2.x. Still the same error.
Tried downgraded Angular CLI from 6.1.x to 6.0.7 and 6.0.8. But still same error.

**Comment by pertrai1** [comment]
@sureshdotariya did you check to make sure that the node_modules did indeed get pruned and there is nothing cached? Possibly blowing away node_modules and doing install again?

**Comment by sureshdotariya** [comment]
Yes, we also did 'npm cache clean --force' before 'npm install'. But no luck its same error.

**Comment by sureshdotariya** [comment]
Finally we able to resolve this issue. Using Angular CLI V6.0.8 with Rxjs V6.3.1.

**Comment by bampakoa** [comment]
I have upgraded to the latest version of RxJS but the problem still persists. Below is my configuration:

Angular CLI: 6.1.5
Node: 8.11.4
OS: win32 x64
Angular: 6.1.6
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.6.3
@angular-devkit/build-angular     0.6.3
@angular-devkit/build-optimizer   0.6.3
@angular-devkit/core              0.7.5
@angular-devkit/schematics        0.7.5
@angular/cdk                      6.4.7
@angular/cli                      6.1.5
@angular/material                 6.4.7
@ngtools/webpack                  6.0.3
@schematics/angular               0.7.5
@schematics/update                0.7.5
rxjs                              6.3.2
typescript                        2.9.2
webpack                           4.8.3

**Comment by pertrai1** [comment]
@bampakoa can you grep your `node_modules` directory for rxjs in all package.json files to see if there are any conflicts in version numbers.

**Comment by cartant** [comment]
@bampakoa You should be able to get npm to list all `rxjs` packages - and their versions - under `node_modules` by running:

```
npm list rxjs
```

**Comment by sureshdotariya** [comment]
I solved the problem by doing these steps

npm install --save-dev @angular/cli@6.0.8
npm cache clean --force

npm install  //this will install rxjs 6.3.2
npm install rxjs-compat@6.2.2

**Comment by bampakoa** [comment]
This is the output from the command as suggested by @cartant and @pertrai1 . Thanks for the help guys!

+-- @angular-devkit/build-angular@0.6.3
| +-- @angular-devkit/architect@0.6.3
| | +-- @angular-devkit/core@0.6.3
| | | `-- rxjs@6.1.0  deduped
| | `-- rxjs@6.1.0
| +-- @angular-devkit/core@0.6.3
| | `-- rxjs@6.1.0  deduped
| +-- @ngtools/webpack@6.0.3
| | `-- @angular-devkit/core@0.6.3
| |   `-- rxjs@6.1.0
| `-- rxjs@6.1.0
+-- @angular/cli@6.1.5
| +-- @angular-devkit/architect@0.7.5
| | `-- rxjs@6.3.2  deduped
| +-- @angular-devkit/core@0.7.5
| | `-- rxjs@6.3.2  deduped
| +-- @angular-devkit/schematics@0.7.5
| | `-- rxjs@6.3.2  deduped
| +-- @schematics/update@0.7.5
| | `-- rxjs@6.3.2  deduped
| `-- rxjs@6.3.2  deduped
`-- rxjs@6.3.2

**Comment by JorgeJoseAbad** [comment]
TypeError: sink._addParentTeardownLogic is not a function solved
#4070 
#12072

**Comment by bampakoa** [comment]
I can confirm that the issue is resolved using Angular CLI 6.2.1

**Comment by jakubjosef** [comment]
Fixed by enforcing 6.3.2 for all dependencies.

Add to package.json
```
  "resolutions": {
    "**/rxjs": "6.3.2"
  },
```
